ui/page/restore.go

```diff
@@ -0,0 +1,446 @@
+							layout.Flex{Axis: layout.Horizontal, Alignment: layout.Middle}.Layout(gtx,
+								layout.Rigid(func() {
+									pg.cancelBtn.Layout(gtx, pg.cancelWdg)
+								}),
+								layout.Rigid(func() {
+									layout.Inset{Left: unit.Dp(10)}.Layout(gtx, func() {
+										gtx.Constraints.Width.Min = 100
+										pg.submitBtn.Layout(gtx, pg.submitWdg)
+									})
+								}),
+							)
+						})
+					}),
+					layout.Rigid(func() {
+						layout.Inset{Top: unit.Dp(15)}.Layout(gtx, func() {
+							pg.confirmErrorMsg.Alignment = text.Middle
+							pg.confirmErrorMsg.Layout(gtx)
+						})
+					}),
+				)
+			})
+		})
+	})
+}
+
+func (pg *Restore) drawModalInputRow(
+	gtx *layout.Context, label string, i *material.Editor, e *widget.Editor) {
+	layout.Flex{Axis: layout.Vertical, Alignment: layout.Baseline}.Layout(gtx,
+		layout.Rigid(func() {
+			pg.theme.Label(unit.Dp(16), label).Layout(gtx)
+		}),
+		layout.Rigid(func() {
+			layout.Inset{Top: unit.Dp(8)}.Layout(gtx, func() {})
+		}),
+		layout.Rigid(func() {
+			i.Layout(gtx, e)
+		}),
+	)
+}
+
+func (pg *Restore) validateWords() bool {
+	for i, editor := range pg.editors {
+		txt := editor.Text()
+		pg.inputLabels[i].Color = BlackColor
+		if txt == "" {
+			pg.inputLabels[i].Color = DangerColor
+			return false
+		}
+	}
+	return true
+}
+
+func (pg *Restore) wordsFromEditors() string {
+	var txt string
+	for _, editor := range pg.editors {
+		txt += editor.Text() + " "
+	}
+	return txt
+}
+
+func (pg *modal) validatePassword() bool {
+	pg.confirmErrorMsg.Text = ""
+	if pg.editorPassword.Text() == "" {
+		pg.confirmErrorMsg.Text = "Please enter password"
+		return false
+	}
+	if pg.editorConfirmPassword.Text() == "" {
+		pg.confirmErrorMsg.Text = "Please enter confirm password"
+		return false
+	}
+	if pg.editorConfirmPassword.Text() != pg.editorPassword.Text() {
+		pg.confirmErrorMsg.Text = "Password and confirm password do not match"
+		return false
+	}
+	return true
+}
+
+func (pg *Restore) cleanAllInputs() {
+	for _, input := range pg.editors {
+		input.SetText("")
+	}
+	pg.editorsEventsHandlerIndex = -1
+	pg.editorPassword.SetText("")
+	pg.editorSendingPin.SetText("")
+	pg.editorConfirmPassword.SetText("")
+}
+
+func (pg *Restore) clicksEventsHandler(gtx *layout.Context) {
+	if pg.backWdg.Clicked(gtx) {
+		pg.pageRedirect = EventNav{
+			Current: RestoreID,
+			Next:    LandingID,
+		}
+		go func() {
+			pg.pageRedirect = nil
+		}()
+	}
+
+	if pg.restoreWdg.Clicked(gtx) && pg.validateWords() {
+		pg.showModal = true
+	}
+
+	if pg.cancelWdg.Clicked(gtx) {
+		pg.showModal = false
+	}
+
+	if pg.submitWdg.Clicked(gtx) && pg.validatePassword() {
+		if pg.isRestoring {
+			return
+		}
+
+		pg.isRestoring = true
+		pg.submitBtn.Text = "Please wait..."
+		pg.submitBtn.Background = pg.theme.Color.Hint
+		pg.wal.RestoreWallet(pg.wordsFromEditors(), pg.editorSendingPin.Text())
+
+		go func() {
+			resp := <-pg.wal.Send
+
+			if resp.Err != nil {
```

Comment:
See #32 for how to handle non-blocking wallet commands. TLDR: page issues the command, window receives the result.